qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/XssOnPhoneNumberFieldSteps.java

```diff
@@ -0,0 +1,124 @@
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.ActiveVisitsPage;
+import org.openmrs.contrib.qaframework.page.RegistrationEditSectionPage;
+
+public class XssOnPhoneNumberFieldSteps  extends Steps {
+    
+    private TestData.PatientInfo patient;
+    private ActiveVisitsPage activeVisitsPage;
+    private RegistrationEditSectionPage registrationEditSectionPage;
+
+    @Before(RunTest.HOOK.SELENIUM_XSS_VULNERABILITY)
+    public void setUp() throws Exception {
+        initiateWithLogin();
+        patient = createTestPatient();
+        new TestData.TestVisit(patient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_XSS_VULNERABILITY)
+    public void tearDown() throws Exception {
+        deletePatient(patient);
+        quit();
+    }  
+    
+    @Given("user goes to active visit search page")
+    public void userGoesToActiveVisitSearch() {
+        activeVisitsPage = homePage.goToActiveVisitsSearch();
+    }
+
+    @When("user searches for patient using patient identifier")
+    public void searchForPatientUsingPatientIdentifier() {
+        activeVisitsPage.search(patient.identifier);
+    }
+
+    @Then("system loads  patient dashboard page of last active visit")
+    public void systemGoesToPatientDashboardOfLastActiveVisit() {
+        dashboardPage = activeVisitsPage.goToPatientDashboardOfLastActiveVisit();
+    }
+    
+    @And("user clicks on show contact link")
+    public void clickOnShowContact() {
+        dashboardPage.clickOnShowContact();
+    }
+
+    @And("user clicks on edit contact info")
+    public void clickOnEditContact() {
+        registrationEditSectionPage = dashboardPage.clickOnEditContact();
+    }
+
+    @And("user clicks on phone number edit")
+    public void userClicksOnPhoneNumberEdit(){
+        registrationEditSectionPage.clickOnPhoneNumberEdit();
+    }
+
+    @And("user clears phone number")
+    public void userClearsPhoneNumber() {
+        registrationEditSectionPage.clearPhoneNumber();
+    }
+
+    @And("user enters new phone number with xss check script")
+    public void userEntersPhoneNumber() {
+        registrationEditSectionPage.enterPhoneNumber("<script>alert(0)</script>");
```

Comment:
@dkayiwa  our focus for this ticket on this line whereby we input a script instead of an integer for validation.i don't want to use any string to validate because it won't bring out the context of checking vulnerability within the input element. so when i call it phoneNumber verification/validation, meaning i can use any string like "aaaa" . which is not case.Am following same work done here https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-legacy-tests/src/test/java/org/openmrs/contrib/qaframework/legacy/XSSOnPhoneNumberFieldTest.java .

Reply:
Our phone number field is not subject to XSS attacks because it accepts only numeric values. So it is useless to test it against XSS attacks.

Reply:
Ok great. can we leave it in selenium legacy part or we can make it like XssOnPhoneNumberField  because i remember we developed this in a form of testing xss on input  fields as another area of concern.

Reply:
Testing for XSS is fine for other fields that accept non numeric inputs but not for this one. So this test is simply about the field not accepting any non numeric input. It has nothing to do with XSS.

Reply:
Ok thanks. will call it phoneNumber validation